agents/agents/guard/fraud.go

```diff
@@ -0,0 +1,792 @@
+		}
+
+		// Check if we should submit the state report.
+		shouldSubmit, err := g.shouldSubmitStateReport(ctx, fraudSnapshot)
+		if err != nil {
+			return fmt.Errorf("could not check if should submit state report: %w", err)
+		}
+		if !shouldSubmit {
+			return nil
+		}
+
+		// Submit the state report to summit.
+		srSignature, _, _, err := state.SignState(ctx, g.bondedSigner)
+		if err != nil {
+			return fmt.Errorf("could not sign state: %w", err)
+		}
+		ok, err := g.prepareStateReport(ctx, fraudSnapshot.Agent, g.summitDomainID)
+		if err != nil {
+			return fmt.Errorf("could not prepare state report on summit: %w", err)
+		}
+		if !ok {
+			continue
+		}
+		_, err = g.txSubmitter.SubmitTransaction(ctx, big.NewInt(int64(g.summitDomainID)), func(transactor *bind.TransactOpts) (tx *ethTypes.Transaction, err error) {
+			tx, err = g.domains[g.summitDomainID].Inbox().SubmitStateReportWithSnapshot(
+				transactor,
+				int64(stateIndex),
+				srSignature,
+				fraudSnapshot.Payload,
+				fraudSnapshot.Signature,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("could not submit state report with snapshot to summit: %w", err)
+			}
+
+			return
+		})
+		if err != nil {
+			return fmt.Errorf("could not submit SubmitStateReportWithSnapshot tx: %w", err)
+		}
+
+		// Submit the state report to the remote chain.
+		ok, err = g.prepareStateReport(ctx, fraudSnapshot.Agent, fraudSnapshot.AgentDomain)
+		if err != nil {
+			return fmt.Errorf("could not prepare state report on summit: %w", err)
+		}
+		if !ok {
+			continue
+		}
+		_, err = g.txSubmitter.SubmitTransaction(ctx, big.NewInt(int64(fraudSnapshot.AgentDomain)), func(transactor *bind.TransactOpts) (tx *ethTypes.Transaction, err error) {
+			tx, err = g.domains[fraudSnapshot.AgentDomain].LightInbox().SubmitStateReportWithSnapshot(
+				transactor,
+				int64(stateIndex),
+				srSignature,
+				fraudSnapshot.Payload,
+				fraudSnapshot.Signature,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("could not submit state report with snapshot to agent domain %d: %w", fraudSnapshot.AgentDomain, err)
+			}
+
+			return
+		})
+		if err != nil {
+			return fmt.Errorf("could not submit SubmitStateReportWithSnapshot tx: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Only submit a state report if we are not on Summit, and the snapshot
+// agent is not currently in dispute.
+func (g Guard) shouldSubmitStateReport(ctx context.Context, snapshot *types.FraudSnapshot) (bool, error) {
+	var disputeStatus types.DisputeStatus
+	var err error
+	contractCall := func(ctx context.Context) error {
+		disputeStatus, err = g.domains[g.summitDomainID].BondingManager().GetDisputeStatus(ctx, snapshot.Agent)
+		if err != nil {
+			return fmt.Errorf("could not get dispute status: %w", err)
+		}
+
+		return nil
+	}
+	err = retry.WithBackoff(ctx, contractCall, g.retryConfig...)
+	if err != nil {
+		return false, fmt.Errorf("could not get dispute status: %w", err)
+	}
+
+	isNotSummit := snapshot.AgentDomain != 0
```

Comment:
Shouldn't this be called `isNotGuard` / `isNotary`?

SynChain doesn't have any assigned Notaries, and Guards are assigned to every chain (they have zero value for domain to showcase that).

Reply:
Makes sense- opted for `isNotary`